tmt/utils.py

```diff
@@ -909,25 +909,28 @@ def ascii(text: Any) -> bytes:
 def listify(
         data: Union[Tuple[Any, ...], List[Any], str, Dict[Any, Any], 'tmt.steps.StepData'],
         split: bool = False,
-        keys: Optional[List[str]] = None) -> Union[List[Any], Dict[Any, Any],
-                                                   'tmt.steps.StepData']:
+        keys: Optional[List[str]] = None) -> Union[List[Any], Dict[Any, Any]]:
     """
     Ensure that variable is a list, convert if necessary
     For dictionaries check all items or only those with provided keys.
     Also split strings on white-space/comma if split=True.
     """
+    # Avoid circular imports
+    from tmt.steps import StepData
+
     separator = re.compile(r'[\s,]+')
     if isinstance(data, tuple):
         data = list(data)
     if isinstance(data, list):
         return fmf.utils.split(data, separator) if split else data
     if isinstance(data, str):
         return fmf.utils.split(data, separator) if split else [data]
+    if isinstance(data, StepData):
+        return data.to_dict()
```

Comment:
I *think* we probably don't need this anymore. `listify(StepData)` was used because `StepData` was a `TypedDict`, i.e. a dictionary hiding behind a fancy type alias. But that changes in this patch, and we probably no longer even call `listify()` for step-data-like objects, because they are now more visible, and use their `to_raw()` or `to_serialized()` methods, for example. Therefore, `listify()` shouldn't need to cover this type at all. Good point, I'll check the code and I expect this to be gone completely.

Reply:
Dropped support for `StepData` in [8d096fd](https://github.com/teemtee/tmt/pull/1461/commits/8d096fd6dedfb1f4dab9cea49f75946075af5836)

Reply:
Yes, thanks for cleaning this.